### ISSUE TITLE
value of Ns in sketch-rnn

### ISSUE THREAD

**Issue by vianamp** [tensorflow/magenta#961]
Page 4 of https://arxiv.org/pdf/1704.03477.pdf says that `Si = (0,0,0,0,1) for i > Ns`. However, I can find the definition of `Ns` neither in the paper or in the code. Any help?

**Comment by hardmaru** [comment]
Hi!

Please take a look at this document:

https://github.com/tensorflow/magenta/tree/master/magenta/models/sketch_rnn

"Each example in the dataset is stored as list of coordinate offsets: ∆x, ∆y, and a binary value representing whether the pen is lifted away from the paper. This format, we refer to as stroke-3, is described in this paper. Note that the data format described in the paper has 5 elements (stroke-5 format), and this conversion is done automatically inside the DataLoader. Below is an example sketch of a turtle using this format:"

Hope that helps!

**Comment by vianamp** [comment]
Hi @hardmaru. Thanks for you comment.

However, I am still confused. In the document you sent, I found `We suggest you build a dataset where the maximum sequence length is less than 250`. Does that mean that `Ns=250`?

Anyways, I am having trouble to set the size of the sequence used as input for training. Do the sequences always have to start with at the beginning of a drawing and finishes at the end of a drawing?

**Comment by vianamp** [comment]
Hi @hardmaru 

Sorry for bothering you again.

I am trying to create a deterministic regressor for the next point of a sketch given a sequence of 16 points. My loss function is composed of two parts. The first (loss_1) takes into account the MSE of true position and predicted position. Second part (loss_2) is a categorical cross entropy that accounts for the length 3 array of possible pen states. As an initial attempt, I am using different weights (1,10,100) to deal with the unbalanced number of possible states in the data. As the training goes on, I see that loss_2 is decaying, but still much greater than loss_1 (~80x) and does not seem to be getting any close to loss_1 in the future. See the figure enclosed.

![image](https://user-images.githubusercontent.com/7592099/33447182-a468f766-d5e9-11e7-8d69-ba97d6e5af3c.png)

Is this a behavior that you would expect? I'd appreciate if you could give me any advice and I'd be happy to provide you more information about the model etc.
